crypto/s2n_dhe.c

```diff
@@ -142,7 +142,7 @@ int s2n_pkcs3_to_dh_params(struct s2n_dh_params *dh_params, struct s2n_blob *pkc
     uint8_t *original_ptr = pkcs3->data;
     dh_params->dh = d2i_DHparams(NULL, (const unsigned char **) (void *) &pkcs3->data, pkcs3->size);
     POSIX_GUARD(s2n_check_p_g_dh_params(dh_params));
-    if (pkcs3->data && (pkcs3->data - original_ptr != pkcs3->size)) {
+    if (pkcs3->data && ((uint32_t) (pkcs3->data - original_ptr) != pkcs3->size)) {
         DH_free(dh_params->dh);
```

Comment:
This is probably safe based on the guarantees of d2i_DHparams (and what are the chances that a signed value happens to wrap back around to the correct size?), but you might want to add a check that original_ptr <= pkcs3->data to make it completely safe.

But that would make the if statement pretty messy:
```suggestion
    if (pkcs3->data && (pcks3->data < original_ptr || ((uint32_t) (pkcs3->data - original_ptr) != pkcs3->size))) {
        DH_free(dh_params->dh);
```

Instead, can you handle this cleanup with a DEFER_CLEANUP? Especially since it already looks like we're not freeing the dh_params if the check on line 144 fails. If you can use DEFER_CLEANUP, you can make individual assertions about the params instead of needing that complicated if statement.